### PR TITLE
correct the catalogsource format

### PR DESCRIPTION
**Pull request by jianzhangbjz**
There are two points of updates on this PR.

- the catalogsource format.
Based on https://redhat-internal.slack.com/archives/C3VS0LV41/p1684941388946749, the `CatalogSource` is 
```yaml 
spec:
    sourceType: grpc
    image: "registry.build01.ci.openshift.org/ci-op-y7mkyv2r/pipeline@sha256:9ec669a1e6cb2a86464a1eeda3e4e478555a3e68dc108158020f613d89e10b70"
    grpcPodConfig:
    securityContextConfig: restricted
CatalogSource name is "oo-wmtz4"
```
But, the generated `catalogsource` resource is
```yaml
spec:
  image: registry.build01.ci.openshift.org/ci-op-rydx97yh/pipeline@sha256:03137438f69bdb20d44469b6c3c4152f0dbdf54592b2ac3e296315cecce1d824
  sourceType: grpc
status:
...
```
As you can see, the  `securityContextConfig` field was omitted. Because its format is incorrect.

- the PSA enabled version.
The PSA default `enable` has been reverted to `privileged` for 4.12 and 4.13. Details:
OCP 4.12:
```yaml
MacBook-Pro:~ jianzhang$ oc get clusterversion
NAME      VERSION   AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.12.19   True        False         85m     Cluster version is 4.12.19
MacBook-Pro:~ jianzhang$ jq "" $(oc extract cm/config -n openshift-kube-apiserver --confirm) | jq '.admission.pluginConfig.PodSecurity'
{
  "configuration": {
    "apiVersion": "pod-security.admission.config.k8s.io/v1beta1",
    "defaults": {
      "audit": "restricted",
      "audit-version": "latest",
      "enforce": "privileged",
      "enforce-version": "latest",
      "warn": "restricted",
      "warn-version": "latest"
    },
    "exemptions": {
      "usernames": [
        "system:serviceaccount:openshift-infra:build-controller"
      ]
    },
    "kind": "PodSecurityConfiguration"
  }
}
```
OCP 4.13
```yaml
MacBook-Pro:~ jianzhang$ oc get clusterversion
NAME      VERSION                              AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.13.0-0.nightly-2023-05-27-155444   True        False         16m     Cluster version is 4.13.0-0.nightly-2023-05-27-155444
MacBook-Pro:~ jianzhang$ 
MacBook-Pro:~ jianzhang$ jq "" $(oc extract cm/config -n openshift-kube-apiserver --confirm) | jq '.admission.pluginConfig.PodSecurity'
{
  "configuration": {
    "apiVersion": "pod-security.admission.config.k8s.io/v1beta1",
    "defaults": {
      "audit": "restricted",
      "audit-version": "latest",
      "enforce": "privileged",
      "enforce-version": "latest",
      "warn": "restricted",
      "warn-version": "latest"
    },
    "exemptions": {
      "usernames": [
        "system:serviceaccount:openshift-infra:build-controller"
      ]
    },
    "kind": "PodSecurityConfiguration"
  }
}
```
OCP 4.14 
```yaml
MacBook-Pro:~ jianzhang$ oc get clusterversion
NAME      VERSION                              AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.14.0-0.nightly-2023-05-28-015206   True        False         30m     Cluster version is 4.14.0-0.nightly-2023-05-28-015206
MacBook-Pro:~ jianzhang$ jq "" $(oc extract cm/config -n openshift-kube-apiserver --confirm) | jq '.admission.pluginConfig.PodSecurity'
{
  "configuration": {
    "apiVersion": "pod-security.admission.config.k8s.io/v1",
    "defaults": {
      "audit": "restricted",
      "audit-version": "latest",
      "enforce": "restricted",
      "enforce-version": "latest",
      "warn": "restricted",
      "warn-version": "latest"
    },
    "exemptions": {
      "usernames": [
        "system:serviceaccount:openshift-infra:build-controller"
      ]
    },
    "kind": "PodSecurityConfiguration"
  }
}
```

### CODE REVIEW

**Comment by jianzhangbjz** [comment]
/pj-rehearse pull-ci-openshift-kni-cluster-group-upgrades-operator-main-integration

File: ci-operator/step-registry/optional-operators/subscribe/optional-operators-subscribe-commands.sh [hunk 1]
```diff
@@ -240,12 +240,12 @@ else
   CS_NAMESPACE="${OO_INSTALL_NAMESPACE}"
 fi
 
-# The securityContextConfig API field was added in 4.12
+# The securityContextConfig API field was added in 4.12, but the default "enforce" is "restricted" since 4.14
 CS_PODCONFIG=""
 OCP_MINOR_VERSION=$(oc version | grep "Server Version" | cut -d '.' -f2)
-if [ "$OCP_MINOR_VERSION" -gt "11" ]; then
+if [ "$OCP_MINOR_VERSION" -gt "13" ]; then
```
Comment (Xia-Zhao-rh) [inline ci-operator/step-registry/optional-operators/subscribe/optional-operators-subscribe-commands.sh:246]: If the featuregate TechPreviewNoUpgrade is enable, the catsrc cannot be READY

**Comment by jianzhangbjz** [comment]
Test pass: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/39799/rehearse-39799-pull-ci-openshift-kni-cluster-group-upgrades-operator-main-integration/1663016809319632896/artifacts/integration/optional-operators-subscribe/build-log.txt

File: ci-operator/step-registry/optional-operators/subscribe/optional-operators-subscribe-commands.sh (hunk 1 above)
Comment (jianzhangbjz) [reply ci-operator/step-registry/optional-operators/subscribe/optional-operators-subscribe-commands.sh:246]: Yes, thanks for the reminder, I will update it later.

Comment (jianzhangbjz) [reply ci-operator/step-registry/optional-operators/subscribe/optional-operators-subscribe-commands.sh]: Add "restricted" since 4.12, thanks!

**Comment by jianzhangbjz** [comment]
Hi @kuiwang02 @Xia-Zhao-rh Could you help re-add `/lgtm` for it? Thanks!

**Comment by jianzhangbjz** [comment]
Hi @dirgim , could you help approve it? Thanks!

**Comment by jianzhangbjz** [comment]
/pj-rehearse pull-ci-openshift-kni-cluster-group-upgrades-operator-main-integration

**Comment by asergienk** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.10-cvp-common-claim

**Comment by asergienk** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.13-cvp-common-claim

**Pull request merged**